### PR TITLE
feat: Handle missing prices in subscription calculator

### PR DESCRIPTION
**Pull request by JesperGomes**

### CODE REVIEW

File: source/databricks/calculation_engine/tests/calculation/wholesale/test_subscription_calculators.py [hunk 1]
```diff
@@ -204,6 +204,89 @@ def test__returns_expected_amount(
...
+            prepared_subscriptions,
+            DefaultValues.CALCULATION_PERIOD_START,
+            DefaultValues.CALCULATION_PERIOD_END,
+            DefaultValues.TIME_ZONE,
+        )
+
+        # Assert
+        assert actual.collect()[0][Colname.charge_count] == expected_charge_count
+        assert actual.collect()[0][Colname.charge_price] == expected_charge_price
+        assert actual.collect()[0][Colname.total_amount] == expected_charge_amount
```
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/tests/calculation/wholesale/test_subscription_calculators.py:247]: How can you be sure that it's the first row?
If there should be only row, should you assert that as well?

File: source/databricks/calculation_engine/tests/calculation/wholesale/test_subscription_calculators.py [hunk 2]
```diff
@@ -204,6 +204,89 @@ def test__returns_expected_amount(
...
+            prepared_subscriptions,
+            DefaultValues.CALCULATION_PERIOD_START,
+            DefaultValues.CALCULATION_PERIOD_END,
+            DefaultValues.TIME_ZONE,
+        )
+
+        # Assert
+        assert actual.collect()[0][Colname.charge_count] == expected_charge_count
+        assert actual.collect()[0][Colname.charge_price] is None
+        assert actual.collect()[0][Colname.total_amount] is None
```
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/tests/calculation/wholesale/test_subscription_calculators.py:287]: How can you be sure that it's the first row?
If there should be only one row, should you assert that as well?

Comment (JesperGomes) [reply source/databricks/calculation_engine/tests/calculation/wholesale/test_subscription_calculators.py:287]: yes, you are right. I will assert that there is only one row

**Review by BjarkeMeier — APPROVED** [review]

File: source/databricks/calculation_engine/package/calculation/wholesale/subscription_calculators.py [hunk 3]
```diff
@@ -16,23 +16,26 @@
...
 from pyspark.sql import DataFrame
 import pyspark.sql.functions as f
+
 from package.constants import Colname
 
 
-def calculate_daily_subscription_amount(
-    subscription_charges: DataFrame,
+def calculate_subscription_amount(
+    prepared_subscriptions: DataFrame,
```
[the remark this answers is not in the document]
Comment (JesperGomes) [reply source/databricks/calculation_engine/package/calculation/wholesale/subscription_calculators.py:24]: Yes, I am thinking the same. I was considering if we could streamline this a bit more as a part of this story

**Pull request merged**